go-update/src/main/java/org/reactome/release/goupdate/GoTermInstanceModifier.java

```diff
@@ -359,13 +363,76 @@ else if (GoTermsUpdater.getReferrerCountsExcludingGOEntities(this.goInstance).is
 			e.printStackTrace();
 		}
 	}
+
+	/*
+	 * Clears reference attributes that point TO *this* goInstance FROM other GO entities. To be used when an instance is being deleted.
+	 */
+	private void clearAttributesFromReferringGOEntities() throws Exception
+	{
+		Map<GKSchemaAttribute, Integer> goReferrerCounts = GoTermsUpdater.getReferrerCountsFilteredByClass(this.goInstance, GoTermsUpdater.isNotGOEntity.negate());
+		// If there are any referrers that are GO entities...
+		if (goReferrerCounts.size() > 0)
+		{
+			for (GKSchemaAttribute attrib : goReferrerCounts.keySet())
+			{
+				// set the referring attributes to NULL so that we don't end up with "dangling pointers" in the database.
+				Collection<GKInstance> attribReferrers = (Collection<GKInstance>) this.goInstance.getReferers(attrib);
+				for (GKInstance attribReferrer : attribReferrers)
+				{
+					try
+					{
+						updatedGOTermLogger.info("CLEARING the attribute {} on \"{}\" because it refers to \"{}\", which is flagged for deletion.", attrib.getName(), abbreviate(attribReferrer.toString(), 55), abbreviate(this.goInstance.toString(), 55));
```

Comment:
Why 55 for the max length?

Reply:
I think that was when I was still trying to determine optimal length for abbreviated strings.